noir-projects/noir-protocol-circuits/crates/rollup-lib/src/abis/block_rollup_public_inputs.nr

```diff
@@ -58,7 +58,7 @@ impl BlockRollupPublicInputs {
         // the indices.
         let num_blocks = self.new_archive.next_available_leaf_index
             - self.previous_archive.next_available_leaf_index;
-        assert(num_blocks < (1 << 16), "Too many accumulated blocks");
+        num_blocks.assert_max_bit_size::<16>();
```

Comment:
So this means there's an allocation of ~65k blocks every checkpoint? Seems nice and generous!